add QA to docs (#1374)
PyTorchLightning/pytorch-lightning: .github/CONTRIBUTING.md

4. Use tags in PR name for following cases:
    - **[blocked by #<number>]** if you work is depending on others changes
    - **[wip]** when you start to re-edit your work, mark it so no one will accidentally merge it in meantime

### Question & Answer

1. **Is there a recommendation for branch names?**
    

>     We do not rely on on the name convention so far you are working with your own fork. Anyway it would be nice to follow this convention `<type>/<issue-id>_<short-name>` where the types are: `bugfix`, `feaure`, `docs`, `tests`, ...

Reviewer: ```suggestion
    We do not rely on the name convention so far you are working with your own fork. Anyway it would be nice to follow this convention `<type>/<issue-id>_<short-name>` where the types are: `bugfix`, `feaure`, `docs`, `tests`, ...
```

> 1. **How to rebase my PR?**

Reviewer: ```suggestion
2. **How to rebase my PR?**
```

>     We recommend to create a PR in separate branch the `master`, especially if you plan to submit several changes and do not want to wait till the fist one is resolved (we can work on them in parallel). Update your master with upstream (assume you have already set [upstream](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork))

Reviewer: ```suggestion
    We recommend to create a PR in separate branch different from `master`, especially if you plan to submit several changes and do not want to wait until the fist one is resolved (we can work on them in parallel). Update your master with upstream (assuming you have already set [upstream](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork))
```

    ```bash
    git fetch --all --prune
    git checkout master
    git merge upstream/master
    ```
    checkout your feature branch
    ```bash
    git checkout my-PR-branch
    git rebase master
    # follow git instructions to resolve conflists
    git push -f

>     ```

Reviewer: ```suggestion
    ```
3. **Should I add type hints in the signature or in the docs, or both?**
    We prefer to have the type hints in the method/function signature.
    They should not be added to the docstring because Sphinx will automatically generate them for us. This avoids duplication. 
```

here is one possible thing we encounter frequently. 
It could also go to the documentation section? what do you think?

Author: this doc is in docs - https://pytorch-lightning.readthedocs.io/en/latest/CONTRIBUTING.html

Reviewer 2: i looked at the link, i don't see a recommendation regarding type hints. am i missing something?

Author: this change is not published yet...
